### ISSUE TITLE
PBS /setuid format type based on sync type

### ISSUE THREAD

**Issue by bretg** [prebid/prebid-server#1831]
Followup to 1554 and 1477

Once PBS supports bidders having both types of usersync (iframe and redirect) (issue #1554) and has updated the /setuid endpoint to accept a format parameter (issue #1477), there's a gap.

The /setuid call needs to have f=i for redirect and f=b for iframes.

There are a couple of ways we could do this:

1) Update all 100+ bidder-configured /setconfig urls to contain the relevant format value. And make sure all future URLs contain the right one as well.
2) Have the PBS /cookie_sync endpoint dynamically update the configured "redirect-url": if it starts with /setuid and doesn't contain an 'f'  parameter, add the correct value ( f=i for redirect and f=b for iframes)

**Comment by SyntaxNode** [comment]
This requirement is actually already included in the description for https://github.com/prebid/prebid-server/issues/1554, bullet point number 5. :)

I'm currently implementing this in PBS-Go as part of an overall user sync rewrite. I like the approach used in PBS-Java to specify the user sync settings in the config file instead of using code. I've iterated on the design a bit, specifically with the redirect-url. The only parts of the redirect-url which need to be different per bidder is the bidder name (we already know this via the config) and the uid macro. I've hoisted the redirect-url to a host level default with each bidder specifying just the macro name. Something like this:

Host Config
```
user-sync-defaults:
  host-url: https:\\some.server.com\prebid
  redirect-url: {{.HostURL}}/setuid?bidder={{.SyncerKey}}&gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&uid={{.UIDMacro}}&f={{.Kind}}
```

Bidder Config
```
user-sync:
  key: adnxs
  default: iframe
  iframe:
    url: http://some.server.com/sync?redirect={{.RedirectURL}}
    redirect-url-macro: $UID
    support-cors: true
  redirect:
    url: http://some.server.com/sync?redirect={{.RedirectURL}}
    redirect-url-macro: $UID
    support-cors: true
```

These settings can all be overridden per bidder using the config system, and eventually per alias too. I haven't yet encountered a use case for overriding the redirect-url, but better to have the flexibility. If the redirect url doesn't need to be used, the macro may simply be omitted and PBS-Go will ignore it.

This approach enables full backwards compatibility with PBS-Go's current config system, where only the url is configured per bidder.

If the setuid `f` parameter is missing from the redirect url or is an empty string, the setuid endpoint will respond the same way as it does now. This use case will be encountered by any host who currently has user sync url overrides. This behavior is intended as a bridge to give hosts time to update, and perhaps even an incentive.

**Comment by bretg** [comment]
> This requirement is actually already included in the description for #1554, bullet point number 5. :)

Right! Just verified this is indeed working in PBS-Java 1.62. 

Your user-sync config update looks reasonable to me. Note that PBS-Java also now supports setting two user-sync urls -- primary and secondary, one redirect one iframe, in either order.
